### PR TITLE
Several fixes / changes to the `holder` item action

### PR DESCRIPTION
**Pull request by apace100**
This PR includes several changes regarding the holder item action:

1. Optimized the `InventoryUtil#forEachStack` method to not allocate a new set each call
2. Changed the setting of holder entities from vanilla's `setHolder` to our duck interface's `EntityLinkedItemStack#setEntity`. This is better because it does not change vanilla semantics of the holder field.
3. Added a test power.
4. Renamed `holder` to `holder_action` and the `entity_action` field to `action`. This is to make it in line with other actions, like `equipped_item_action`, `passenger_action`, etc. I feel this is okay since the action hasn't been released for too long and is probably still barely used. If you think this is too late for that change, we could add an alias using the old names for backwards compatibility and make an announcement asking people to change over to the new names quickly.

Thanks Pug for pointing out the optimization potential with the method.

### CODE REVIEW

File: src/main/java/io/github/apace100/apoli/util/InventoryUtil.java [hunk 1]
```diff
@@ -386,13 +389,25 @@ public static void forEachStack(Entity entity, Consumer<ItemStack> itemStackCons
     }
 
     private static void deduplicateSlots(Entity entity, Set<Integer> slots) {
```
Comment (MerchantPug) [inline src/main/java/io/github/apace100/apoli/util/InventoryUtil.java:391]: Are we sure that this method should be kept here? We aren't using it anymore and it also isn't exposed for anybody else to use, so I feel like we could remove it.

**Review by ChrysanthCow — CHANGES_REQUESTED** [review]
~~I don't quite see the point in keeping `InventoryUtil#deduplicateSlots` when we aren't using it anymore and when it's private.~~

~~If it is still used, please tell me and I'll approve. I can't quite check properly on my phone.~~

~~I feel like we could potentially implement these optimisations to the other `InventoryUtil` methods too.~~

File: src/main/java/io/github/apace100/apoli/util/InventoryUtil.java (hunk 1 above)
Comment (MerchantPug) [reply src/main/java/io/github/apace100/apoli/util/InventoryUtil.java:391]: There's two things we can do here as this method **is** used.

1. Ignore this review.
2. Implement the optimisations to the set creation inside the other `InventoryUtil` methods.

**Review by ChrysanthCow — APPROVED** [review]
The above is invalid, please ignore it.

**Review by eggohito — APPROVED** [review]
Looks good! I agree on adding a path alias for it, though identifier aliasing is currently not perfect (it doesn't have a priority system of some sort similar to `Prioritized.CallInstance<?>`)

**Comment by ChrysanthCow** [comment]
> Looks good! I agree on adding a path alias for it, though identifier aliasing is currently not perfect (it doesn't have a priority system of some sort similar to `Prioritized.CallInstance<?>`)

Agreed on having a  path alias.

**Pull request merged**